ledger/block/src/transaction/deployment/mod.rs

```diff
@@ -124,6 +124,11 @@ impl<N: Network> Deployment<N> {
         &self.verifying_keys
     }
 
+    /// Returns the total number of constraints.
+    pub fn num_constraints(&self) -> u64 {
+        self.verifying_keys.iter().map(|(_, (vk, _))| vk.circuit_info.num_constraints).sum::<usize>() as u64
```

Comment:
```suggestion
        self.verifying_keys.iter().map(|(_, (vk, _))| vk.circuit_info.num_constraints as u64).sum::<u64>()
```
just to limit the possibility of an overflow in `u32` architectures

Reply:
6b6fc407a